File: 7-kyu/maximum-length-difference/maximum-length-difference.js
function mxdiflg(a1, a2) {
  if (!a1.length || !a2.length) return -1;
  let max = 0;
  for(let x of a1){
    for(let y of a2){
      max = Math.max(max, Math.abs(x.length - y.length));
    } 
  }
  return max;
}

File: 7-kyu/maximum-length-difference/maximum-length-difference.py
def mxdiflg(a1, a2): # O(n+m)
    if not a1 or not a2:
        return -1
    len1 = [len(s) for s in a1]
    len2 = [len(s) for s in a2]
    return max(abs(max(len1) - min(len2)), abs(max(len2) - min(len1)))